fabtests/prov/efa/src/efa_info_test.c

```diff
@@ -37,13 +37,47 @@
 
 int main(int argc, char **argv)
 {
-	int ret;
+	int ret = 0, op;
 	struct fi_info *info;
 
 	hints = fi_allocinfo();
 	if (!hints)
 		return EXIT_FAILURE;
-	hints->fabric_attr->prov_name=strdup("efa");
+
+
+	while ((op = getopt(argc, argv, "f:p:")) != -1) {
+		switch (op) {
+		case 'f':
+			if (!hints->fabric_attr) {
```

Comment:
I thought `fi_allocinfo()` already allocated `fabric_attr`, see https://github.com/ofiwg/libfabric/blob/main/src/fabric.c#L1425-L1446

Reply:
Yes, I think the fabtests code were written to handle the case that application alloc hints manually via malloc. OFI 2.0 requires that all hints must be allocated via `fi_dupinfo(NULL)` (which is the same as `fi_allocinfo`).